### DISCUSSION TITLE
Deployment as standalone app on linux server

### DISCUSSION THREAD

**Discussion by lpj145** [loco-rs/loco#324] [General]
After all, big thank you guys for you efforts on this, really, this is very very fascinating.

**Description**
I'd like to deploy my app as standalone https server on linux in some place of the earth, something like small business do: deploy a linux server configured with cloudflare and server yours product solutions.

**To Reproduce**

So bad so true, I can't see any place that I can configure my app, by the way, searching in the loco source code I saw that [line](https://github.com/loco-rs/loco/blob/c06223c2486dfb1db5bf9586d2b15c3ad6c6c969/src/boot.rs#L167) whats turn things out a little bad.

**Expected Behavior**

Server standalone application with ssl

**Additional Context**

I can handle this, can I provide the pr?

**Comment by kaplanelad** [comment]
@lpj145 moving this thread to discussion.

What do you mean by configuring your app?  

if I understand correctly, you have a server that runs your binary app. When the process is running, the app listens on your given port. 
if you want to expose your server externally with SSL you can use nginx or you have a vendor cloud like AWS that can do it for you

**Reply by lpj145** [reply]
I mean, I want to run it directly on linux with a ssl without any vendor on top of that, just the binary and ssl and eveything is fine.

**Reply by jondot** [reply]
I would strongly recommend having ngnix on that Linux server, to handle SSL termination (we have deploy generator for ngnix). Would that work for you?

**Reply by kaplanelad** [reply]
You can start your loco server; when it's up, it exposes you to a port on the Linux machine (like you saw in the link).
Afterward, navigate to your Linux IP/DNS. For example, if your server is assigned the IP address x.x.x and you execute the Loco process on port 3000, you can access your application by visiting x.x.x:3000 (without SSL).

In case you want SSL you need configure it in your Linux server

**Reply by lpj145** [reply]
Guys I respect your point of view, I know how valuable the nginx could be, but, trust me, I only want to have nothing on top of my service, by the way and if I want too not have nothing more than my final binary on the pod image? it increases the chance to up some pods with ssl in handful double ssl handshake between my load balance or reverse proxy and my pod. In this link we can see how axum could implement tls [here](https://github.com/rustls/hyper-rustls/blob/def00ca3d4d9dca46bfde5574f4b0f3d2b722a74/examples/server.rs#L40) and [here](https://github.com/tokio-rs/axum/blob/f7c4cd883a2548c23b60e2ee952a33e7ca85fbe1/examples/low-level-rustls/src/main.rs#L46) the axum uses hyper under the hood.

**Reply by kaplanelad** [reply]
@lpj145, now your case is more clear for me :) 

We are not supporting this right now, and we can add it. 
You can open us an issue. you can also open us if you want

**Reply by jondot** [reply]
Just to add my 2cent, we always try to simplify, adding SSL termination and handling code will add complexity (loading certificates, specialized binding, cert pinning, and other edge cases) to the boot hot path and to documentation, which is why at this stage we prefer not to support built-in SSL termination.
